### ISSUE TITLE
Remove Singularity Install/Config

### ISSUE THREAD

**Issue by dvdgomez** [canonical/ov-slurm-charms#4]
Singularity is currently installed and configured using juju actions. Remove these actions such that the functionality can instead be separated into an individual singularity charm. Slurm_ops_manager seems to be tied to each install and will need to be removed wrt to installs as seen below.

## Remove Resources

- [ ] Remove .deb/.rpm resources

Remove from charm-slurmd/metadata.yaml

## Remove Actions

- [ ] singularity-install

Remove from charm-slurmd/actions.yaml

**Comment by NucciTheBoss** [comment]
Yes, we should remove the Singularity actions, but I think we should at least have a published Apptainer package somewhere before we do that. I have potential snap package we could use [here](https://github.com/canonical/apptainer), however, it needs some more work before we can release it publicly on the Snap Store.

We should also look into producing a deb package just in case we have issues propagating the Apptainer snap across shared file systems. The Apptainer developers do have a working Debian recipe, but they need someone to undertake the work to get the package into the official Debian/Ubuntu repositories. I do have a PPA setup [here](https://launchpad.net/~nuccitheboss/+archive/ubuntu/apptainer) but I need some time to allocate towards creating the package.

**Comment by NucciTheBoss** [comment]
It also looks like someone is working towards releasing Apptainer in an official PPA, but I have not seen any mention of this effort on the Apptainer GitHub: https://launchpad.net/~apptainer/+archive/ubuntu/ppa. The single person in the team however has not listed a public email address.
